src/Core/System/Test/SystemConfig/Command/ConfigSetCommandTest.php

```diff
@@ -0,0 +1,70 @@
+<?php declare(strict_types=1);
+
+namespace Shopware\Core\System\Test\SystemConfig\Command;
+
+use PHPUnit\Framework\TestCase;
+use Shopware\Core\Framework\Test\TestCaseBase\IntegrationTestBehaviour;
+use Shopware\Core\System\SystemConfig\Command\ConfigSet;
+use Shopware\Core\System\SystemConfig\SystemConfigService;
+use Shopware\Core\Test\TestDefaults;
+use Symfony\Component\Console\Tester\CommandTester;
+
+class ConfigSetCommandTest extends TestCase
+{
+    use IntegrationTestBehaviour;
+
+    private ConfigSet $configSetCommand;
+    private SystemConfigService $systemConfigService;
+
+    protected function setUp(): void
+    {
+        $this->systemConfigService = $this->createMock(SystemConfigService::class);
+        $this->configSetCommand = new ConfigSet($this->systemConfigService);
+    }
+
+    public function configValueProvider(): iterable
+    {
+        /* config_key, value, expected_value, decode */
+        yield 'String false' => ['fake_config_key', 'false', 'false', false];
+        yield 'Decode boolean false' => ['fake_config_key', 'false', false, true];
+        yield 'String int' => ['fake_config_key', '4', '4', false];
+        yield 'Decode String int' => ['fake_config_key', '5', 5, false];
+        yield 'String float' => ['fake_config_key', '2.2', '2.2', false];
+        yield 'Decode String float' => ['fake_config_key', '3.3', 3.3, false];
+        yield 'String json' => [
+            'fake_config_key',
+            "{\"name\":\"abc\",\"place\":\"xyz\"}",
+            "{\"name\":\"abc\",\"place\":\"xyz\"}",
+            false,
+        ];
+        yield 'Decode String json' => [
```

Comment:
i also would add a test case that some random string will be passed as is even if the decode option is enabled
```php
yield 'Decode string remains string' => [
    'random string',
    'random string',
     true,
];
```

Reply:
Done.